include/libtorrent/aux_/unique_ptr.hpp

```diff
@@ -50,6 +50,9 @@ namespace libtorrent { namespace aux {
 		using base = std::unique_ptr<T[]>;
 		using underlying_index = typename underlying_index_t<IndexType>::type;
 
+		unique_ptr() {}
+		explicit unique_ptr(T arr[]) : base(arr) {}
```

Comment:
in fact, I think it would be better to just pull in the base class' constructors, instead of defining new ones.

I believe this should do it: ``using base::base;``